src/Rings/groebner.jl

```diff
@@ -419,29 +422,297 @@ Singular Polynomial Ring (QQ),(a,b,c),(dp(3),C)
 julia> I = Singular.Ideal(SR,[SR(-1+c+b+a^3),SR(-1+b+c*a+2*a^3),SR(5+c*b+c^2*a)])
 Singular ideal over Singular Polynomial Ring (QQ),(a,b,c),(dp(3),C) with generators (a^3 + b + c - 1, 2*a^3 + a*c + b - 1, a*c^2 + b*c + 5)
 
-julia> Oscar.normal_form_internal(I,J)
+julia> Oscar.normal_form_internal(I,J,default_ordering(base_ring(J)))
 3-element Vector{fmpq_mpoly}:
  a^3
  2*a^3 + 2*a - 2*c
  4*a - 2*c^2 - c + 5
 ```
 """
-function normal_form_internal(I::Singular.sideal, J::MPolyIdeal)
-  groebner_assure(J)
-  G = collect(values(J.gb))[1]
-  singular_assure(G)
+function normal_form_internal(I::Singular.sideal, J::MPolyIdeal, o::MonomialOrdering)
+  groebner_assure(J, o)
+  G = J.gb[o]  
+  singular_assure(G, o)
   K = ideal(base_ring(J), reduce(I, G.S))
   return [J.gens.Ox(x) for x = gens(K.gens.S)]
 end
 
 @doc Markdown.doc"""
-    normal_form(f::T, J::MPolyIdeal) where { T <: MPolyElem }
+	reduce(I::IdealGens, J::IdealGens; ordering::MonomialOrdering = default_ordering(base_ring(J)))
+
+Return a `Vector` whose elements are the underlying elements of `I`
```

Comment:
Maybe explain reduce(f,F;...) first, and then rely on this for explaining the entries of the returned vector in the other two cases. Right now the explanations for reduce (F,G;..) and reduce(I,J;...) are correct and complete, but rather hard to read.